### PR TITLE
Fixes the typing rule of CAUGHTadm

### PR DESCRIPTION
**Pull request by ioannad**
Addresses review comment on merged #229: 
https://github.com/WebAssembly/exception-handling/pull/229#discussion_r1029160012

### CODE REVIEW

**Comment by ioannad** [comment]
I just realised there are some more comments on this in open PR #241, which I will address in this PR (tomorrow). I will make all changes to the typing of CAUGHTadm here, because they don't fit in #241.

File: document/core/appendix/properties.rst [hunk 1]
```diff
@@ -676,25 +676,28 @@ To that end, all previous typing judgements :math:`C \vdash \X{prop}` are genera
...
+* The label :math:`C.\CLABELS[0]` must be defined in the context.
 
-* Under context :math:`C'`,
-  the instruction sequence :math:`\instr^\ast` must be :ref:`valid <valid-instr-seq>` with type :math:`[] \to [t^\ast]`.
+* Let :math:`(\LCATCH^?~[t^\ast])` be the :ref:`label type <syntax-labeltype>` :math:`C.\CLABELS[0]`.
+
+* Let :math:`C'` be the same :ref:`context <context>` as :math:`C`, but with the first label popped from the |CLABELS| vector.
 
-* Let :math:`C''` be the same :ref:`context <context>` as :math:`C`, but with the label :math:`[t^\ast]` prepended to the |CLABELS| vector.
+* Let :math:`C''` be the same :ref:`context <context>` as :math:`C'`, but with the label type :math:`(\LCATCH~[t^\ast])` prepended to the |CLABELS| vector.
```
Comment (rossberg) [inline document/core/appendix/properties.rst]: I think you can merge these:

```suggestion
* Let :math:`C''` be the same :ref:`context <context>` as :math:`C`, but with the label type :math:`(\LCATCH~[t^\ast])` replacing the first element of the |CLABELS| vector.
```

File: document/core/appendix/properties.rst [hunk 2]
```diff
@@ -676,25 +676,28 @@ To that end, all previous typing judgements :math:`C \vdash \X{prop}` are genera
...
    \frac{
      S \vdashexternval \EVTAG~\tagaddr : \ETTAG~[t_0^\ast]\to[]
      \qquad
      (val : t_0)^\ast
      \qquad
-     S; C,\CLABELS\,(\LCATCH~[t^\ast]) \vdashinstrseq \instr^\ast : [] \to [t^\ast]
+     S; C',\CLABELS\,(\LCATCH~[t^\ast]) \vdashinstrseq \instr^\ast : [] \to [t^\ast]
    }{
-     S; C,\CLABELS\,[t^\ast] \vdashadmininstr \CAUGHTadm\{\tagaddr~\val^\ast\}~\instr^\ast~\END : [] \to [t^\ast]
+     S; C',\CLABELS\,(\LCATCH^?~[t^\ast]) \vdashadmininstr \CAUGHTadm\{\tagaddr~\val^\ast\}~\instr^\ast~\END : [] \to [t^\ast]
```
Comment (rossberg) [inline document/core/appendix/properties.rst]: I don't think we can or want to allow this to be a catch label beforehand already.
```suggestion
     S; C',\CLABELS\,[t^\ast] \vdashadmininstr \CAUGHTadm\{\tagaddr~\val^\ast\}~\instr^\ast~\END : [] \to [t^\ast]
```

**Review by rossberg — APPROVED** [review]
LGTM modulo suggestions.

File: document/core/appendix/properties.rst (hunk 1 above)
Comment (ioannad) [reply document/core/appendix/properties.rst]: For some reason I couldn't apply the suggestions but I added these changes to the latest commit I just pushed.

File: document/core/appendix/properties.rst (hunk 2 above)
Comment (ioannad) [reply document/core/appendix/properties.rst]: Same as above

**Comment by ioannad** [comment]
Applied the suggestions manually, it seemed impossible otherwise. Merging.

**Pull request merged**